packages/gatsby-source-filesystem/src/gatsby-node.js

```diff
@@ -196,7 +212,15 @@ See docs here - https://www.gatsbyjs.org/packages/gatsby-source-filesystem/
 
   return new Promise((resolve, reject) => {
     watcher.on(`ready`, () => {
-      fsMachine.send({ type: `CHOKIDAR_READY`, resolve, reject })
+      const { unsubscribe } = fsMachine.subscribe(state => {
+        // TODO: decide when to reject
+        if (state.matches(`CHOKIDAR.READY`)) {
+          unsubscribe()
+          resolve()
+        }
+      })
+
+      fsMachine.send({ type: `CHOKIDAR_READY` })
```

Comment:
`CHOKIDAR.READY` and `CHOKIDAR_READY` Is hard to distinct. And the next one debugging is wondering why point or interline or is this an error. Better a litle more different names.

Reply:
Good point, I'll think about a differentiating them.